scripts/tools/InitializeQueue.php

```diff
@@ -103,6 +112,10 @@ public function __invoke($params)
                             $broker = new RdsQueueBroker($this->persistenceId, $this->receive ?: 1);
                             break;
 
+                        case self::BROKER_NEW_SQL:
```

Comment:
As an improvement and to reinforce the open/closed I would extract this switch case to a class (or at least a method) to return a `QueueBrokerInterface`. This list of brokers is growing and this class is no longer just initializing parameters. 

A suggestion:

```
class BrokerFactory
{
    public const BROKER_MEMORY = 'memory';
    public const BROKER_RDS = 'rds';
    public const BROKER_NEW_SQL = 'newsql';
    public const BROKER_SQS = 'sqs';

    public function create(string $brokerId, string $persistenceId, int $capacity = 1): QueueBrokerInterface
    {
        switch ($brokerId) {
            case self::BROKER_MEMORY:
                return new InMemoryQueueBroker();
            case self::BROKER_RDS:
                return new RdsQueueBroker($persistenceId, $capacity);
            case self::BROKER_NEW_SQL:
                return new NewSqlQueueBroker($persistenceId, $capacity);
            case self::BROKER_SQS:
                return new SqsQueueBroker(\common_cache_Cache::SERVICE_ID, $capacity);
        }

        throw new \InvalidArgumentException(sprintf('Broker %s is not supported', $brokerId));
    }
}
```

I think it is easy change and since we are touching this class, we could do it :)

Reply:
done